dapps-certification-persistence/src/IOHK/Certification/Persistence/Structure/Subscription.hs

```diff
@@ -0,0 +1,232 @@
+  { featureId :: FeatureType
+  , featureName :: Text
+  } deriving (Generic, Show)
+
+dropAndLowerFirst :: Int -> String ->  String
+dropAndLowerFirst n = toLowerFirst . drop n
+  where
+  toLowerFirst :: String -> String
+  toLowerFirst [] = []
+  toLowerFirst (x:xs) = Char.toLower x : xs
+
+instance ToJSON Feature where
+  toJSON = genericToJSON defaultOptions { fieldLabelModifier = dropAndLowerFirst 7 }
+
+instance FromJSON Feature where
+  parseJSON = genericParseJSON defaultOptions { fieldLabelModifier = dropAndLowerFirst 7 }
+
+instance ToSchema Feature where
+  declareNamedSchema = genericDeclareNamedSchema defaultSchemaOptions { fieldLabelModifier = dropAndLowerFirst 7 }
+
+instance SqlRow Feature
+
+features :: Table Feature
+features = tableFieldMod "feature"
+  [ #featureId :- primary
+  , #featureName :- unique
+  ] (fromJust . stripPrefix "feature")
+
+data Tier = Tier
+  { tierId :: ID Tier
+  , tierName :: Text
+  , tierSubtitle :: Text
+  , tierDescription :: Text
+  , tierType :: TierType
+  , tierUsdPrice :: Double
+  , tierDuration :: Int
+  , tierEnabled :: Bool
+  } deriving (Generic, Show)
+
+type TierId = ID Tier
+
+instance ToJSON Tier where
+  toJSON = genericToJSON defaultOptions { fieldLabelModifier = dropAndLowerFirst 4 }
+
+instance FromJSON Tier where
+  parseJSON = genericParseJSON defaultOptions { fieldLabelModifier = dropAndLowerFirst 4 }
+
+instance ToSchema Tier where
+  declareNamedSchema = genericDeclareNamedSchema defaultSchemaOptions { fieldLabelModifier = dropAndLowerFirst 4 }
+
+instance ToSchema TierId where
+  declareNamedSchema _ = declareNamedSchema (Proxy :: Proxy Int64)
+
+instance FromJSON (ID Tier) where
+  parseJSON = withText "ID Tier" $ \t -> pure $ toId $ read $ Text.unpack t
+
+
+instance ToJSON (ID Tier) where
+  toJSON = toJSON . show . fromId
+instance ToParamSchema TierId where
+  toParamSchema _ = mempty
+    & type_ ?~ SwaggerInteger
+
+instance SqlRow Tier
+
+tiers :: Table Tier
+tiers = tableFieldMod "tier"
+  [ #tierId :- autoPrimary
+  ] (fromJust . stripPrefix "tier")
+
+data TierFeature = TierFeature
+  { tierFeatureId :: ID TierFeature
+  , tierFeatureTierId :: ID Tier
+  , tierFeatureFeatureId :: FeatureType
+  } deriving (Generic, Show)
+
+instance SqlRow TierFeature
+
+tierFeatures :: Table TierFeature
+tierFeatures = tableFieldMod "tier_feature"
+  [ #tierFeatureId :- autoPrimary
+  , #tierFeatureTierId :- foreignKey tiers #tierId
+  , #tierFeatureFeatureId :- foreignKey features #featureId
+  ] (fromJust . stripPrefix "tierFeature")
+
+data SubscriptionStatus = ActiveSubscription | InactiveSubscription | PendingSubscription
+                       deriving (Generic, Show, Bounded, Enum, Read)
+data Subscription = Subscription
+  { subscriptionId :: ID Subscription
+  , subscriptionProfileId :: ID Profile
+  , subscriptionTierId :: ID Tier
+  , subscriptionName :: Text
+  , subscriptionType :: TierType
+  , subscriptionPrice :: Int64
```

Comment:
No, this is the amount of Lovelace to be paid for the subscription in order to be activated. We try to keep as `long` any Lovelace amounts since they tend to be quite long